specs/pvs/corla.pvs

```diff
@@ -1650,6 +1761,186 @@ END system_assumptions
+%%% data. For example, they define who is on their audit boards, they
+%%% generate digests of files critical to the audit (principly, ballot
+%%% manifests and CVRs), they upload those files, etc.
+%%%
+%%% For general county information, we currently model the data as
+%%% property/value pairs in a record. For example, we imagine that
+%%% each county will want to display who their County Clerk is, etc.
+
+  county_general_information_table: TYPE =
+    sequence[[# property: string, value: string #]]
+  audit_board_table: TYPE =
+    sequence[[county, audit_board_member]]
+  ballot_manifest_digest_table: TYPE =
+    sequence[[# county: county,
+                ballot_manifest_file_name: string,
+                digest: string,
+                data: ballot_manifest_file #]]
+  cvr_digest_table: TYPE =
+    sequence[[# county: county,
+                cvr_file_name: string,
+                digest: string,
+                data: cvr_file #]]
+  ballot_manifest_table: TYPE =
+    sequence[[county, ballot_manifest_info]]
+
+%%% Cast vote records uploaded by counties contain CVRs of three
+%%% kinds: CVRs for local (county) contests, CVRs for contests that
+%%% span counties, and CVRs for contests that span the whole
+%%% state. These latter two cases are identical from a modeling point
+%%% of view, as the whole state is simply a spanning contest over all
+%%% counties.
+%%%
+%%% We can store and organize this information in several different
+%%% ways. First, we can simply store the individual CVRs uploaded by
+%%% a given county in a table.
+
+  cvr_table: TYPE =
+    sequence[[county, cvr]]
+
+%%% But we also need to store all CVRs for each contest, aggregating
+%%% all CVRs across the upload from all counties in which the contest
+%%% was on the ballot.
+
+  contest_cvr_table: TYPE =
+    sequence[[contest, cvr]]
+
+%%% Finally, the tabulation results are uploaded by the counties, and
+%%% is the key information necessary in order to run the RLA
+%%% algorithm.
+
+  tabulation_result_table: TYPE =
+    sequence[[contest, rla_tabulation_outcome]]
+
+%%% While running the RLA, the Audit Board uploads information about
+%%% each ballot they audit. Also, if any mistakes are made and
+%%% remedied during the audit, that information must be stored as
+%%% well. Finally, after an audit is complete an audit report is
+%%% uploaded and stored.
+
+  ballots_under_audit_table: TYPE =
+    sequence[[ballot, ballot_manifest_info, list[ballot_mark],
+              cvr, margin_overstatement, margin_understatement]]
+  auditing_mistakes_table: TYPE =
+    sequence[[county, audit_mistake_report]]
+  audit_reports_table: TYPE =
+    sequence[[county, audit_report]]
+  rla_summary_results_table: TYPE =
+    sequence[[contest, summary_results_report]]
+
+%%% All of the information on the public dashboard is derived from the
+%%% above data.
+
+% @todo kiniry Add election as first column to every table? Or have a
+% different DB per election?
```

Comment:
Either is likely to work fine given the data volume we're talking about here. Having a separate DB per election has appeal because of the otherwise unbounded growth of those tables over time, and for ease of archiving.